source/D2Common/src/D2Inventory.cpp

```diff
@@ -3264,69 +3264,54 @@ BOOL __stdcall INVENTORY_RemoveAllItems(D2InventoryStrc* pInventory)
 	return TRUE;
 }
 
-//D2Common.0x6FD921D0 (#10302)
-BOOL __stdcall INVENTORY_CanItemsBeTraded(void* pMemPool, D2UnitStrc* pPlayer1, D2UnitStrc* pPlayer2, BOOL* a4)
+// Helper function
+BOOL __fastcall INVENTORY_CanItemBePlacedInTradeInventory(D2InventoryStrc* pTradeInventory, D2UnitStrc* pPlayer1, D2UnitStrc* pPlayer2)
 {
-	BOOL bResult = FALSE;
-	
+	for (D2UnitStrc* pItem = INVENTORY_GetFirstItem(pPlayer1->pInventory); pItem; pItem = INVENTORY_GetNextItem(pItem))
+	{
+		pItem = INVENTORY_UnitIsItem(pItem);
```

Comment:
Should `INVENTORY_UnitIsItem` be renamed to `INVENTORY_ItemCast` ?

Reply:
Hmm, I'd say it's not really a cast because it just checks if it's an item and then returns it or nullptr.

Reply:
Well true, but in C++ a dynamic cast can fail for example, but I don't really care about the name that much anyway :)

Reply:
This isn't a cast; the check vs the unit type is done to prevent items that are invalid (being deleted) from being interacted with.